apps/web/src/pages/ProfileAndApplicationsPage/ProfileAndApplicationsPage.tsx

```diff
@@ -68,14 +68,29 @@ const ProfileAndApplicationsApplicant_Query = graphql(/* GraphQL */ `
   }
 `);
 
+export const loader = (client: Client) => async () => {
+  return await client
+    .query(ProfileAndApplicationsApplicant_Query, {})
+    .toPromise()
+    .then((res) => res.data)
+    .then((res) => {
+      if (!res?.me) {
+        return Promise.reject(new NotFoundError());
+      }
+
+      return Promise.resolve(res);
+    });
+};
+
```

Comment:
Added in an example of how this could be done with data loaders. The `then` can be merged into a single one. But we can reject the promise to have it show 404 and not need the `ThrowNotFound` anymore :thinking: